stdlib/public/Concurrency/TaskGroup.swift

```diff
@@ -33,52 +33,45 @@ extension Task {
 }
 
 
-/// Starts a new task group which provides a scope in which a dynamic number of
-/// tasks may be spawned.
+/// Starts a new scope in which a dynamic number of tasks can be spawned.
 ///
-/// Tasks added to the group by `group.spawn()` will automatically be awaited on
-/// when the scope exits. If the group exits by throwing, all added tasks will
-/// be cancelled and their results discarded.
+/// When the group returns,
+/// it implicitly waits for all spawned tasks to complete.
+/// The tasks are canceled only if `cancelAll()` was invoked before returning,
+/// if the groups' task was canceled,
+/// or if the group body throws an error.
+/// ◊TR: Is it possible to throw an error here?
 ///
-/// ### Implicit awaiting
-/// When the group returns it will implicitly await for all spawned tasks to
-/// complete. The tasks are only cancelled if `cancelAll()` was invoked before
-/// returning, the groups' task was cancelled, or the group body has thrown.
-///
-/// When results of tasks added to the group need to be collected, one can
-/// gather their results using the following pattern:
+/// To collect the results of tasks that were added to the group,
+/// use the following pattern:
 ///
 ///     while let result = await group.next() {
-///       // some accumulation logic (e.g. sum += result)
+///         // some accumulation logic (e.g. sum += result)
 ///     }
 ///
 /// It is also possible to collect results from the group by using its
 /// `AsyncSequence` conformance, which enables its use in an asynchronous for-loop,
 /// like this:
 ///
 ///     for await result in group {
-///       // some accumulation logic (e.g. sum += result)
+///         // some accumulation logic (e.g. sum += result)
 ///      }
+/// ◊TR: Which of the above do we prefer?  Why give two recommendations?
```

Comment:
the group.next() can be useful if you want to specifically pick a few tasks, say "pick first two tasks, then cancel the rest"

```
let firstCompleted = await group.next() ?? 0
let secondCompleted = await group.next() ?? 0
group.cancelAll() // cancel the rest
return firstCompleted + secondCompleted
```

while the `for await` is sugar for "keep collecting until next returns nil" (so until we drained all tasks)

Reply:
When i worked on those docs first we did not have the `for await` yet so I only "added" it later.

Perhaps we should do the following here:
- show the `for await` version and recommend it
- "if you need more control i.e. one-by-one picking completed tasks from the group you can use group.next()" and give a `guard let first = await group.next() else { ... }` example?

That should explain when to use that I hope :)